c++/src/Reader.cc

```diff
@@ -1113,7 +1115,11 @@ namespace orc {
                                                    sargsApplier->getNextSkippedRows());
         previousRow = firstRowOfStripe[currentStripe] + currentRowInStripe - 1;
         if (currentRowInStripe > 0) {
-          seekToRowGroup(static_cast<uint32_t>(currentRowInStripe / footer->rowindexstride()));
+          auto rowIndexStride = footer->rowindexstride();
+          auto rowGroupId = static_cast<uint32_t>(currentRowInStripe / rowIndexStride);
+          uint64_t rowsToSkip = currentRowInStripe % rowIndexStride;
+          seekToRowGroup(rowGroupId);
+          reader->skip(rowsToSkip);
```

Comment:
Is it better to call skip() out of this scope since the naming of current function is startNextStripe(). It should only happen when seekToRow() is called. Or we can add an input parameter to startNextStripe() advising the target row to start.

Reply:
OK, I moved the changes to seekToRow(). Also added some test coverage on the case when the file doesn't have row indexes.